### DISCUSSION TITLE
Adicione mais "estilo" ao seu console application!

### DISCUSSION THREAD

**Discussion by niziul** [balta-io/2801#45] [Perguntas e Respostas] [answered]
Queira trazer uma dica rápida para quem tá entusiasmado, como eu estou, pelo tanto de conhecimento que é repassado nesse curso. 

Hoje comecei a assistir ao módulo onde se fala um pouco mais a respeito das [Strings](https://balta.io/player/assistir/5ef614da-366d-6156-d049-694800000000/85dc8974-34be-47a7-a5be-000028010901), e executando a aplicação resolvi "fuçar" um pouco na class `System.Console` e acabei tendo uma ideia para criar um simples sistema que armazenava o id gerado, em um simples arquivo de texto, porém na hora de exibir os ids não ficava do jeito que me agradava, então resolvi criar um interface que fosse responsiva para o console. 

+ Após algumas horas de "funça, funça pra cá, funça, funça pra lá" acabei achando o [`Console.BufferWidth`](https://docs.microsoft.com/en-us/windows/console/console-screen-buffers) não sei ao certo para o que ele é usado, mas me serviu como orientação para saber a largura da área de input disponível no console. Com isso construir essa função:
   
   ![](https://i.ibb.co/4Y5kgXf/Menu-Skeleton.png)

         static void MenuSkeleton()
         {
             for (var i = 0; i <= Console.BufferWidth - 3; i++)
                 Console.Write("-");
         }

+ é algo bem simples, mas eu acredito que pode ajuda quem curte padronizar e/ou destacar na hora de exibir alguma informação no console.

```

```
+ Depois de um tempo cheguei nessa solução onde é possível centralizar qualquer conteúdo no console

        struct Attributes
        {
            public static int DifferenceForCentralization = 2;
        }
        static void MenuSkeleton(string nameForMenu)
        {
            var textForMenu = nameForMenu;

            for (var i = 0; i <= (Console.BufferWidth - textForMenu.Length - Attributes.DifferenceForCentralization) / 2; i++)
                Console.Write("-");

            Console.Write(textForMenu);

            for (var i = (Console.BufferWidth - textForMenu.Length - Attributes.DifferenceForCentralization) / 2; 
                    i <= Console.BufferWidth - textForMenu.Length - Attributes.DifferenceForCentralization; i++)
                Console.Write("-");
        }
        static void WriteCentered(string stringb)
        {
            var textForWrite = stringb;
            for (var i = 0; i <= (Console.BufferWidth - textForWrite.Length - Attributes.DifferenceForCentralization) / 2; i++)
                Console.Write(" ");

            Console.WriteLine(textForWrite);         
        }

+ Criei esse perfil 

        public static void Show()
        {
            Console.Clear();
            MenuSkeleton("<<<<<::>>>>>");
            WriteCentered("[Testing something]");
            MenuSkeleton("<<<<<<<<<<<<<<::>>>>>>>>>>>>>>");
            
        }

![](https://i.ibb.co/ZJt27JG/imagem-2022-02-12-225809.png)

Caso alguém venha a testar, poderia me dizer se ocorreu algum erro?

**Comment by andrebaltieri** [comment, marked as answer]
@NiziulLuizin muito obrigado pelas dicas 💜

**Comment by alonso-boj** [comment]
Muito legal! Eu também acho divertida a ideia de customizar aplicações console e descobri essa biblioteca aqui: [Spectre.Console](https://spectreconsole.net/). Depois dá uma olhada!

**Reply by niziul** [reply]
Muito obrigado pela recomendação. Acabei de dar uma olhada e fiquei realmente impressionado.
